api/tests/Feature/PoolCandidateUpdateTest.php

```diff
@@ -1229,4 +1229,123 @@ public function testManualStatusUpdatePolicy(): void
                 ],
             ]);
     }
+
+    /**
+     * @dataProvider nullTimeProvider
+     */
+    public function testNullTimestampsOnStatusChanged($status, $expected)
+    {
+        $past = config('constants.past_datetime');
+        $this->poolCandidate->pool_candidate_status = PoolCandidateStatus::DRAFT->name;
+        $this->poolCandidate->submitted_at = $past;
+        $this->poolCandidate->removed_at = $past;
+        $this->poolCandidate->final_decision_at = $past;
+        $this->poolCandidate->placed_at = $past;
+        $this->poolCandidate->save();
+
+        $this->actingAs($this->communityAdminUser, 'api')
+            ->graphQL($this->manualStatusUpdateMutation, [
+                'id' => $this->poolCandidate->id,
+                'candidate' => ['status' => $status],
+            ])->assertJsonFragment([
+                'status' => [
+                    'value' => $status,
+                ],
+                ...$expected,
+            ]);
+    }
+
+    public static function nullTimeProvider()
```

Comment:
We don't just null timestamps we also set some. It might be nice to either update this or add a mirrored test that timestamps **are** set when desired.

Reply:
We have that test already. However, it is now incorrect and I am working on trying to fix it.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/8d1557735efd095e1362407fc44c8720e9e11f17/api/tests/Feature/PoolCandidateUpdateTest.php#L990

Reply:
:facepalm: I clearly need to pay more attention to what else is in these files.

Reply:
And it should be fixed now after [fix test for timestamp update](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12734/commits/42ed5114c7e80eda65e3e6d8fc8bd9bd2ee19c25)

Reply:
It is a large file :laughing: I honeslty was about to add those tests in when I notice that test existed and the only reason I did notice is because it was failing :stuck_out_tongue:

Reply:
I think you're missing SCREENED_OUT_NOT_INTERESTED here.

Reply:
Sorry, I maybe shouldn't have added that here. I meant in testManualStatusUpdatesTimestamps.